realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/mongodb/RemoteMongoCollectionTest.kt

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.mongodb
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import io.realm.*
+import io.realm.log.LogLevel
+import io.realm.log.RealmLog
+import io.realm.mongodb.remote.RemoteCountOptions
+import io.realm.util.blockingGetResult
+import org.bson.Document
+import org.bson.types.ObjectId
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import kotlin.test.assertEquals
+import kotlin.test.assertNull
+
+@RunWith(AndroidJUnit4::class)
+class RemoteMongoCollectionTest {
+
+    private lateinit var app: TestRealmApp
+    private lateinit var user: RealmUser
+    private lateinit var client: RemoteMongoClient
+    private lateinit var database: RemoteMongoDatabase
+
+    @Before
+    fun setUp() {
+        Realm.init(InstrumentationRegistry.getInstrumentation().targetContext)
+        RealmLog.setLevel(LogLevel.DEBUG)
+        app = TestRealmApp()
+        user = app.registerUserAndLogin(TestHelper.getRandomEmail(), "123456")
+        client = user.getRemoteMongoClient(SERVICE_NAME)
+        database = client.getDatabase(DATABASE_NAME)
+    }
+
+    @After
+    fun tearDown() {
+        // FIXME: probably not the best way to "reset" the state
+        with(getCollectionInternal(COLLECTION_NAME)) {
+            deleteMany(Document()).blockingGetResult()
+        }
+
+        if (this::app.isInitialized) {
+            app.close()
+        }
+    }
+
+    @Test
+    fun insertMany() {
```

Comment:
Yeah, this is the first iteration. It has been a bit hard to produce exception scenarios given that the parser isn't 100% ready yet. Also, my approach to this is to port the existing tests on collections from the Stitch SDK. I guess it will be clearer once the parser is fully fixed.